### PR TITLE
[in Graphics][bugfix] Protection against division by zero in TGaxis

### PR DESCRIPTION
**Pull request by andriish**
Dear @couet,

here is a small fix to protect against division by zero in TGaxis.
I saw it once during debug session with master branch on linux.

Best regards,
Andrii

Tag @couet

### CODE REVIEW

**Comment by pcanal** [comment]
@phsft-bot build

File: graf2d/graf/src/TGaxis.cxx [hunk 1]
```diff
@@ -1974,7 +1974,7 @@ void TGaxis::PaintAxis(Double_t xmin, Double_t ymin, Double_t xmax, Double_t yma
                      if (fNModLabs) ChangeLabelAttributes(k+1, nlabels, textaxis, chtemp);
                      typolabel = chtemp;
                      if (!optionTime) typolabel.ReplaceAll("-", "#minus");
-                     if (autotoff) {
+                     if (autotoff && (gPad->GetWw()*gPad->GetWNDC() != 0.0)) {
```
Comment (Axel-Naumann) [inline graf2d/graf/src/TGaxis.cxx:1977]:
```suggestion
                     double scale = gPad->GetWw()*gPad->GetWNDC();
                     if (autotoff && (scale != 0.0)) {
```
and then re-use `scale` in the call to `TMath::Max`.

Comment (Axel-Naumann) [reply graf2d/graf/src/TGaxis.cxx:1977]: Also, I would have assumed that the proper change would have been the call to `TMath::Max`:
```
if (scale > 0.)
   toffset = TMath::Max(toffset,(double)w/scale);
```
Why do you need to protect the whole branch, i.e. why is `scale == 0.0` equivalent to `!autotoff`?

**Review by Axel-Naumann — CHANGES_REQUESTED** [review]

File: graf2d/graf/src/TGaxis.cxx (hunk 1 above)
Comment (andriish) [reply graf2d/graf/src/TGaxis.cxx]: Done.

>the whole branch
That is an "optimization".  My understanding was that the actions in the branch were doene only to calculate the `toffset`. So I've assumed `scale=0` eliminates the need to do that. But your suggestion is better.

Best regards,

Andrii

**Comment by andriish** [comment]
Hi @pcanal ,

I've resolved the thread as the requested changes were implemented.


Andrii

**Review by Axel-Naumann — APPROVED** [review]

**Comment by Axel-Naumann** [comment]
Thanks, @andriish !

**Comment by andriish** [comment]
Thank you, @Axel-Naumann ,

would you have a look at the other requests from me? All of them are trivial, many are approved.
There is no urgency, just hope these will be accepted before 6.24.

Best regards,

Andrii

**Pull request merged**